net-p2p/bitcoin-core/bitcoin-core-25.1.ebuild

```diff
@@ -0,0 +1,340 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..12} )
+
+inherit autotools bash-completion-r1 db-use desktop python-any-r1 systemd xdg-utils
+
+DESCRIPTION="Reference implementation of the Bitcoin cryptocurrency"
+HOMEPAGE="https://bitcoincore.org/"
+SRC_URI="https://bitcoincore.org/bin/${P}/${P/-core}.tar.gz"
+S="${WORKDIR}/${P/-core}"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux"
+# IUSE="+cli" doesn't work due to https://bugs.gentoo.org/831045#c3
+IUSE="+asm +berkdb +bitcoin-cli +daemon dbus examples +external-signer kde libs +man nat-pmp +qrcode qt5 +sqlite system-leveldb +system-libsecp256k1 systemtap test upnp zeromq"
+RESTRICT="!test? ( test )"
+
+REQUIRED_USE="
+	dbus? ( qt5 )
+	kde? ( qt5 )
+	qrcode? ( qt5 )
+	system-leveldb? ( || ( daemon qt5 ) )
+"
+# dev-libs/univalue is now bundled, as upstream dropped support for system copy
+# and their version in the Bitcoin repo has deviated a fair bit from upstream.
+# Upstream also seems very inactive.
+RDEPEND="
+	!<dev-util/bitcoin-tx-25.0
+	>=dev-libs/boost-1.81.0:=
+	>=dev-libs/libevent-2.1.12:=
+	berkdb? ( >=sys-libs/db-4.8.30:$(db_ver_to_slot 4.8)=[cxx] )
+	bitcoin-cli? ( !<net-p2p/bitcoin-cli-25.0 )
+	daemon? (
+		!<net-p2p/bitcoind-25.0
+		acct-group/bitcoin
+		acct-user/bitcoin
+	)
+	libs? ( !<net-libs/libbitcoinconsensus-25.0 )
+	nat-pmp? ( >=net-libs/libnatpmp-20220705:= )
+	qrcode? ( >=media-gfx/qrencode-4.1.1:= )
+	qt5? (
```

Comment:
This should be gui per qa policy if it's the only flag which causes a graphical interface.

Reply:
I have noticed a lot of packages changing their `IUSE="qt5"` to `IUSE="gui"` in recent months. That strikes me as regressive, as it _reduces_ the expressiveness of the user's global `USE` setting. It becomes no longer possible to express, "I want GUIs but only if they are implemented in Qt5; please no Gtk+ on my system," or similar.

Reply:
Rationale is at https://projects.gentoo.org/qa/policy-guide/use-flags.html#pg0802 (on mobile so brief comments, sorry)

Reply:
Thank you. Kind of amusing that the rationale given there is the logical converse of my objection. I personally would _never_ wish to express, "Give me _any_ GUI, no matter how much crap you have to pull in," but I _would_ wish to express, "I'll take a GUI if you can provide it using only Qt5." Oh well; I can't fight Gentoo dogma.

Reply:
The issue is, with the previous state, you end up with USE=qt5 in make.conf doing things like pulling in a GUI for CMake where most people don't really care at all.

I suppose it shines mostly where there's a tool that is very common and has a GUI which most people don't want.